core/src/banking_stage.rs

```diff
@@ -803,11 +807,25 @@ impl BankingStage {
     }
 }
 
+#[cfg_attr(feature = "dev-context-only-utils", qualifiers(pub))]
+pub(crate) fn update_bank_forks_and_poh_recorder_for_new_tpu_bank(
```

Comment:
refer to https://github.com/anza-xyz/agave/pull/3946/files#r1886311350 for justification of this change.